lib/form_data.js

```diff
@@ -144,22 +146,35 @@ FormData.prototype._multiPartHeader = function(field, value, options) {
   if (options.header != null) {
     header = options.header;
   } else {
-    header += '--' + boundary + FormData.LINE_BREAK +
+    header = '--' + boundary + FormData.LINE_BREAK +
       'Content-Disposition: form-data; name="' + field + '"';
 
     // fs- and request- streams have path property
     // or use custom filename and/or contentType
     // TODO: Use request's response mime-type
     if (options.filename || value.path) {
-      header +=
-        '; filename="' + path.basename(options.filename || value.path) + '"' + FormData.LINE_BREAK +
-        'Content-Type: ' +  (options.contentType || mime.lookup(options.filename || value.path));
-
+      filename = path.basename(options.filename || value.path);
+      contentType = (
+        options.contentType ||
+        mime.lookup(options.filename || value.path) ||
+        'application/octet-stream'
+      );
     // http response has not
     } else if (value.readable && value.hasOwnProperty('httpVersion')) {
-      header +=
-        '; filename="' + path.basename(value.client._httpMessage.path) + '"' + FormData.LINE_BREAK +
-        'Content-Type: ' + value.headers['content-type'];
+      filename = path.basename(value.client._httpMessage.path);
+      contentType = value.headers['content-type'] || 'application/octet-stream';
+    } else if (Buffer.isBuffer(value)) {
+      contentType = contentType || 'application/octet-stream';
+    }
+    
+    // Add optional filename.
+    if (filename) {
+      header += '; filename="' + filename + '"';
+    }
+    
+    // Add optional contentType.
+    if (contentType) {
+      header += FormData.LINE_BREAK + 'Content-Type: ' + contentType;
```

Comment:
This part looks fragile. If we leave string manipulation to the last moment, it will be more manageable and less error prone.

Something like:

```
headers['Content-Disposition'] = ['form-data', 'name="' + field + '"'];

if (filename) {
  headers['Content-Disposition'].push('filename="' + filename + '"');
}

if (contentType) {
  headers['Content-Type'] = contentType;
}
```

And then if header is array, join it via `;`.